### PR TITLE
Expose the number of postgres connections

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: sql/src/main/java/io/crate/metadata/sys/SysNodesTableInfo.java [hunk 1]
```diff
@@ -369,24 +373,41 @@ protected BytesRef valueForItem(FsInfo.Path input) {
...
+                        Columns.CONNECTIONS_HTTP_TOTAL.path().get(1),
+                        NestableCollectExpression.<NodeStatsContext, HttpStats>withNullableProperty(
+                            NodeStatsContext::httpStats,
+                            HttpStats::getTotalOpen)
+                    )
+                ),
+                Columns.CONNECTIONS_PSQL.path().get(0),
+                new ObjectCollectExpression<NodeStatsContext>(
+                    ImmutableMap.of(
+                        Columns.CONNECTIONS_HTTP_OPEN.path().get(1),
```
Comment (matriv) [inline sql/src/main/java/io/crate/metadata/sys/SysNodesTableInfo.java]: PSQL

File: sql/src/main/java/io/crate/metadata/sys/SysNodesTableInfo.java [hunk 2]
```diff
@@ -369,24 +373,41 @@ protected BytesRef valueForItem(FsInfo.Path input) {
...
+                        NestableCollectExpression.<NodeStatsContext, ConnectionStats>withNullableProperty(
+                            NodeStatsContext::psqlStats,
+                            ConnectionStats::open),
+                        Columns.CONNECTIONS_HTTP_TOTAL.path().get(1),
```
Comment (matriv) [inline sql/src/main/java/io/crate/metadata/sys/SysNodesTableInfo.java]: Same here

File: sql/src/test/java/io/crate/expression/reference/sys/node/NodeStatsContextFieldResolverTest.java [hunk 3]
```diff
@@ -108,6 +110,20 @@ public void testConnectionsHttpLookupAndExpression() {
...
+        NodeStatsContext statsContext = resolver.forTopColumnIdents(
+            Collections.singletonList(SysNodesTableInfo.Columns.CONNECTIONS));
+        RowCollectExpressionFactory<NodeStatsContext> expressionFactory =
+            SysNodesTableInfo.expressions().get(SysNodesTableInfo.Columns.CONNECTIONS);
+        NestableCollectExpression<NodeStatsContext, ?> expression = expressionFactory.create();
+
+        NestableCollectExpression psql = (NestableCollectExpression) expression.getChild("psql");
+        NestableCollectExpression open = (NestableCollectExpression) psql.getChild("open");
+        open.setNextRow(statsContext);
+        assertThat(open.value(), is(2L));
```
Comment (matriv) [inline sql/src/test/java/io/crate/expression/reference/sys/node/NodeStatsContextFieldResolverTest.java]: For completeness the "total" assertion is missing.

**Comment by mfussenegger** [comment]
@matriv pushed a fixup

**Comment by matriv** [comment]
sorry @mfussenegger, I have one more comment: Shouldn't we test the null -> 0 cases https://github.com/crate/crate/pull/7418/files#diff-396783ee82c43241187c1372f9e460d3R255 ?

**Comment by mfussenegger** [comment]
What do you want to test about it? The static analysis reports a possible NPE without the safe-guard.

**Comment by matriv** [comment]
OK. I would add the `@Nullable` here: https://github.com/crate/crate/pull/7418/files#diff-396783ee82c43241187c1372f9e460d3R95 just for extra visibility
